warp/tests/test_augmented_assign.py

```
@@ -0,0 +1,184 @@
+def test_non_atomic_subscript_sub_single_eval(test, device):
+    """arr[i] -= expr for non-atomic type should evaluate expr exactly once."""
+    counter = wp.zeros(1, dtype=int, device=device)
+    data = wp.array([10.0], dtype=float, device=device)
+
+    wp.launch(kernel_unsupported_op_sub, dim=1, inputs=[counter, data], device=device)
+
+    test.assertEqual(counter.numpy()[0], 1, "RHS evaluated more than once for array -=")
+    test.assertAlmostEqual(data.numpy()[0], 7.0, places=5, msg="data[0] should be 10.0 - 3.0 = 7.0")
+
+
+devices = get_test_devices()
```

Ruff (0.15.5)

[error] `get_test_devices` may be undefined, or defined from star imports

(F405)